pkg/webhook/injection.go

```diff
@@ -71,6 +71,37 @@ func injectSidecarContainer(pod *corev1.Pod, config *Config, supportsNativeSidec
 	}
 }
 
+func (si *SidecarInjector) injectMetadataPrefetchSidecarContainer(pod *corev1.Pod, config *Config, supportsNativeSidecar bool) {
```

Comment:
we should skip injection if the container is already injected. There's no need to do complex logic to reinject. if its injected, skip.

Reply:
Done.